### PR TITLE
Checkout 'Target branch" when dependent PR has already been merged

### PR DESCRIPTION
**Pull request by MichelleXiaoxiaoWang**
- The issue is when a PR has already been merged
- the merge ref we attempt to checkout no longer exists
- check out the json file of pull request
- find the target branch of the merged PR from json file
[skip ci]
Fixes eclipse/openj9#2366

Signed-off-by: Xiaoxiao Wang michellejavajs@gmail.com

### CODE REVIEW

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

File: buildenv/jenkins/common/build [hunk 1]
```diff
@@ -195,26 +196,50 @@ def checkout_pullrequest() {
...
                 sh "git config remote.origin.url https://github.com/eclipse/omr.git"
             }
-            checkout_pullrequest(OMR_PR)
+            checkout_pullrequest(OMR_PR,'omr')
         }
     }
 }
 
-def checkout_pullrequest(ID) {
+def checkout_pullrequest(ID,repo) {
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: Please use all UPPERCASE variable names for consistency

File: buildenv/jenkins/common/build [hunk 2]
```diff
@@ -195,26 +196,50 @@ def checkout_pullrequest() {
...
     // Determine if we are checking out a PR or a branch
+    def JSONFILE = ""
+    def base1 = ""
+	def URL = "https://api.github.com/repos/eclipse/${repo}/pulls/${ID}"
+	withCredentials([usernamePassword(credentialsId: 'b6987280-6402-458f-bdd6-7affc2e360d4', passwordVariable: 'PASS', usernameVariable: 'USER')]) {
+ 		JSONFILE = sh returnStdout: true, script: "curl -u ${USER} --pass  ${PASS} ${URL}"
+	}
+	def slurper1 = new groovy.json.JsonSlurper()
+	def result1 = slurper1.parseText(JSONFILE)
+	base1 = result1.base.ref
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: Uppercase variable names. And perhaps a more descriptive variable name?

File: buildenv/jenkins/common/build [hunk 3]
```diff
@@ -195,26 +196,50 @@ def checkout_pullrequest() {
...
         }
     }
 }
 
-def checkout_pullrequest(ID) {
+def checkout_pullrequest(ID,repo) {
     // Determine if we are checking out a PR or a branch
+    def JSONFILE = ""
+    def base1 = ""
+	def URL = "https://api.github.com/repos/eclipse/${repo}/pulls/${ID}"
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: `eclipse` is the org for openj9 and omr but for the openjdk extensions repos it would be `ibmruntimes`.

File: buildenv/jenkins/common/build [hunk 4]
```diff
@@ -195,26 +196,50 @@ def checkout_pullrequest() {
...
+	withCredentials([usernamePassword(credentialsId: 'b6987280-6402-458f-bdd6-7affc2e360d4', passwordVariable: 'PASS', usernameVariable: 'USER')]) {
+ 		JSONFILE = sh returnStdout: true, script: "curl -u ${USER} --pass  ${PASS} ${URL}"
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: We only need to run this Curl to determine the TARGET_BRANCH of the PR if the PR is already merged. Therefore I think we should move it down into the `catch` block to avoid running it unnecessarily.

File: buildenv/jenkins/common/build [hunk 5]
```diff
@@ -195,26 +196,50 @@ def checkout_pullrequest() {
...
+	def slurper1 = new groovy.json.JsonSlurper()
+	def result1 = slurper1.parseText(JSONFILE)
+	base1 = result1.base.ref  
     try {
         if (ID.isNumber()) {
-            sh "git fetch --progress origin +refs/pull/${ID}/merge:refs/remotes/origin/pr/${ID}/merge"
-            sh "git checkout refs/remotes/origin/pr/${ID}/merge"
+            try {    	
+        		def IDmerge = "${ID}/merge"
+	            sh "git fetch --progress origin +refs/pull/${ID}/merge:refs/remotes/origin/pr/${ID}/merge"
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: Could we use the `fecthRef` function here too?

**Comment by MichelleXiaoxiaoWang** [comment]
screen shot 2018-07-25 at 4 17 41 pm
screen shot 2018-07-25 at 4 17 59 pm

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
#### **1. Correct**
```
17:41:21 Running in /home/jenkins/jenkins-agent/workspace/PullRequest-Compile-JDK10-linux_ppc-64_cmprssptrs_le-OpenJ9/openj9
[Pipeline] {

[Pipeline] sh

17:41:22 [openj9] Running shell script
17:41:22 + git fetch --progress origin +refs/pull/2415/merge:refs/remotes/origin/pr/2415/merge
17:41:22 remote: Counting objects: 11, done.        
17:41:22 remote: Compressing objects:  14% (1/7)           
remote: Compressing objects:  28% (2/7)           
remote: Compressing objects:  42% (3/7)           
remote: Compressing objects:  57% (4/7)           
remote: Compressing objects:  71% (5/7)           
remote: Compressing objects:  85% (6/7)           
remote: Compressing objects: 100% (7/7)           
remote: Compressing objects: 100% (7/7), done.        
17:41:22 remote: Total 11 (delta 4), reused 6 (delta 4), pack-reused 0        
17:41:22 From https://github.com/eclipse/openj9
17:41:22  * [new ref]         refs/pull/2415/merge -> origin/pr/2415/merge
[Pipeline] sh

17:41:23 [openj9] Running shell script
17:41:24 + git checkout refs/remotes/origin/2415/merge
17:41:24 error: pathspec 'refs/remotes/origin/2415/merge' did not match any file(s) known to git.
[Pipeline] withCredentials
[Pipeline] {
[Pipeline] sh

17:41:24 [openj9] Running shell script
17:41:25 + curl -u **** --pass **** https://api.github.com/repos/eclipse/openj9/pulls/2415
17:41:25 Enter host password for user '****':
17:41:25   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
17:41:25                                  Dload  Upload   Total   Spent    Left  Speed
17:41:25 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 18667  100 18667    0     0  50738      0 --:--:-- --:--:-- --:--:-- 50863
[Pipeline] }
[Pipeline] // withCredentials

[Pipeline] sh
17:41:26 [openj9] Running shell script
17:41:27 + git fetch --progress origin +refs/heads/master:refs/remotes/origin/master

[Pipeline] sh
17:41:27 [openj9] Running shell script

17:41:28 + git checkout refs/remotes/origin/master
17:41:28 Note: checking out 'refs/remotes/origin/master'.
17:41:28 
17:41:28 You are in 'detached HEAD' state. You can look around, make experimental
17:41:28 changes and commit them, and you can discard any commits you make in this
17:41:28 state without impacting any branches by performing another checkout.
17:41:28 
17:41:28 If you want to create a new branch to retain commits you create, you may
17:41:28 do so (now or later) by using -b with the checkout command again. Example:
17:41:28 
17:41:28   git checkout -b <new-branch-name>
17:41:28 
17:41:28 HEAD is now at 6ee077f... Merge pull request #2473 from keithc-ca/spec-version
[Pipeline] }
[Pipeline] // dir
[Pipeline] dir
17:41:28 Running in /home/jenkins/jenkins-agent/workspace/PullRequest-Compile-JDK10-linux_ppc-64_cmprssptrs_le-OpenJ9/omr
[Pipeline] {
[Pipeline] sh
```
```
17:41:29 [omr] Running shell script
17:41:29 + git config remote.origin.url https://github.com/eclipse/omr.git
[Pipeline] sh

17:41:30 [omr] Running shell script
17:41:31 + git fetch --progress origin +refs/pull/2771/merge:refs/remotes/origin/pr/2771/merge
17:41:31 fatal: Couldn't find remote ref refs/pull/2771/merge
[Pipeline] withCredentials
[Pipeline] {

[Pipeline] sh
17:41:32 [omr] Running shell script
17:41:32 + curl -u **** --pass **** https://api.github.com/repos/eclipse/omr/pulls/2771
17:41:32 Enter host password for user '****':
17:41:32   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
17:41:32                                  Dload  Upload   Total   Spent    Left  Speed
17:41:32 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 20300  100 20300    0     0  55621      0 --:--:-- --:--:-- --:--:-- 55769
[Pipeline] }
[Pipeline] // withCredentials

[Pipeline] sh
17:41:33 [omr] Running shell script

17:41:34 + git fetch --progress origin +refs/heads/master:refs/remotes/origin/master
[Pipeline] sh

17:41:35 [omr] Running shell script
17:41:35 + git checkout refs/remotes/origin/master
17:41:35 Note: checking out 'refs/remotes/origin/master'.
17:41:35 
17:41:35 You are in 'detached HEAD' state. You can look around, make experimental
17:41:35 changes and commit them, and you can discard any commits you make in this
17:41:35 state without impacting any branches by performing another checkout.
17:41:35 
17:41:35 If you want to create a new branch to retain commits you create, you may
17:41:35 do so (now or later) by using -b with the checkout command again. Example:
17:41:35 
17:41:35   git checkout -b <new-branch-name>
17:41:35 
17:41:35 HEAD is now at b28174b... Merge pull request #2791 from fjeremic/deprecate-unused-api
[Pipeline] }
```

File: buildenv/jenkins/common/build (hunk 5 above)
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/common/build]: fixed

File: buildenv/jenkins/common/build (hunk 4 above)
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/common/build]: fixed

File: buildenv/jenkins/common/build (hunk 3 above)
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/common/build]: fixed

File: buildenv/jenkins/common/build (hunk 2 above)
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/common/build]: fixed

File: buildenv/jenkins/common/build (hunk 1 above)
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/common/build]: fixed

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

File: buildenv/jenkins/common/build [hunk 6]
```diff
@@ -195,26 +196,50 @@ def checkout_pullrequest() {
...
+            try {
+                //sh "git fetch --progress origin +refs/pull/${ID}/merge:refs/remotes/origin/pr/${ID}/merge"
+                fetchRef("pull/${ID}/merge", "pr/${ID}/merge" )
+                checkoutRef ("pr/${ID}/merge")
+            } catch (e) {
+                def JSONFILE = ""
+                def BASEREF = ""
+                def URL = "https://api.github.com/repos/${REPO}/pulls/${ID}"
+                withCredentials([usernamePassword(credentialsId: "${GITHUB_API_CREDENTIALS_ID}", passwordVariable: 'PASS', usernameVariable: 'USER')]) {
+                    JSONFILE = sh returnStdout: true, script: "curl -u ${USER} --pass  ${PASS} ${URL}"
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: 2 whitespaces after `--pass`

File: buildenv/jenkins/common/build [hunk 7]
```diff
@@ -178,15 +179,15 @@ def checkout_pullrequest() {
     }
 
     if (openjdk_bool) {
-        checkout_pullrequest(OPENJDK_PR)
+        checkout_pullrequest(OPENJDK_PR, "ibmruntimes/openj9-openjdk-jdk${SDK_VERSION}")
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: This should fail now that we support JDK(next). I think SDK_VERSION is `next` but the repo is `ibmruntimes/openj9-openjdk-jdk`. Perhaps we need...
```
if (SDK_VERSION == "next") {
    CHECKOUT_REPO = "ibmruntimes/openj9-openjdk-jdk"
} else {
    CHECKOUT_REPO = "ibmruntimes/openj9-openjdk-jdk${SDK_VERSION}"
}
checkout_pullrequest(OPENJDK_PR, CHECKOUT_REPO)
```

**Comment by MichelleXiaoxiaoWang** [comment]
-when a PR has already been merged, the merge ref we attempt to checkout no longer exists and the checkout will fail.
- Use git API call to get the json of pull request and their base branch.
- And a checkout from a merged branch would be to fallback to checkout its base branch.
Fixes eclipse/openj9#2366

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile aix jdk10 depends eclipse/omr#123

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

File: buildenv/jenkins/common/build [hunk 8]
```diff
@@ -178,15 +179,21 @@ def checkout_pullrequest() {
     }
 
     if (openjdk_bool) {
-        checkout_pullrequest(OPENJDK_PR)
+        def CHECKOUT_REPO = ""
+        if (SDK_VERSION == "next") {
+             CHECKOUT_REPO = "ibmruntimes/openj9-openjdk-jdk"
+        } else {
+             CHECKOUT_REPO = "ibmruntimes/openj9-openjdk-jdk${SDK_VERSION}"
+        }
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: This if-else could be trimmed.
```
def REPO_SUFFIX = ""
if (SDK_VERSION != "next") {
    REPO_SUFFIX = SDK_VERSION 
}
checkout_pullrequest(OPENJDK_PR, "ibmruntimes/openj9-openjdk-jdk${REPO_SUFFIX}"
```

File: buildenv/jenkins/common/variables-functions [hunk 9]
```diff
@@ -384,7 +393,6 @@ def set_job_names() {
 def set_job_variables(job_type) {
     // check mandatory parameters if any
     try {
-        validate_arguments(ARGS)
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/variables-functions]: Why was this line removed?

File: buildenv/jenkins/common/variables-functions [hunk 10]
```diff
@@ -469,6 +477,10 @@ def set_user_credentials() {
     if (!USER_CREDENTIALS_ID) {
          USER_CREDENTIALS_ID = ''
     }
+     GITHUB_API_CREDENTIALS_ID = get_github_user_credentials_id()
+     if (!GITHUB_API_CREDENTIALS_ID) {
+         GITHUB_API_CREDENTIALS_ID = ''
+    }
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/variables-functions]: This is technically only required in the "pull-request" case at the moment. I was wondering if we should separate it and move it there but I suppose it's not very much overhead to leave it set in all builds. I guess we can leave it.

File: buildenv/jenkins/common/build (hunk 8 above)
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/common/build]: fixed

File: buildenv/jenkins/common/variables-functions (hunk 9 above)
Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/common/variables-functions]: fixed

File: buildenv/jenkins/common/build [hunk 11]
```diff
@@ -178,15 +179,20 @@ def checkout_pullrequest() {
     }
 
     if (openjdk_bool) {
-        checkout_pullrequest(OPENJDK_PR)
+        def CHECKOUT_REPO = ""
```
Comment (AdamBrousseau) [inline buildenv/jenkins/common/build]: This variable is unused now and can be removed

**Comment by MichelleXiaoxiaoWang** [comment]
- The issue is when a PR has already been merged
- the merge ref we attempt to checkout no longer exists
- check out the json file of pull request
- find the target branch of the merged PR from json file

[skip ci]
Fixes eclipse/openj9#2366

Signed-off-by: Xiaoxiao Wang <michellejavajs@gmail.com>

**Comment by MichelleXiaoxiaoWang** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2771

**Comment by AdamBrousseau** [comment]
Test case: Depends on a merged PR (checks out TARGET_BRACNH)
```
Look for Dependent Changes
PullRequest source repo appears to be OpenJ9
Dependent OMR change found: #2771
+ bash ./get_source.sh
[2018-08-22 14:20:37] Get OpenJ9 sources
Clone repository: omr
Clone repository: openj9
[openj9] Running shell script
+ git fetch --progress origin +refs/pull/2415/merge:refs/remotes/origin/pr/2415/merge
 * [new ref]         refs/pull/2415/merge -> origin/pr/2415/merge
[openj9] Running shell script
+ git checkout refs/remotes/origin/pr/2415/merge
You are in 'detached HEAD' state.
HEAD is now at f64bea0... Merge 9d994e6e31f92170c3fcc7ff37fb243d60df880e into 767109edbe9fd2003476d65cab3a7d30a1ec943d
[omr] Running shell script
+ git config remote.origin.url https://github.com/eclipse/omr.git
[omr] Running shell script
+ git fetch --progress origin +refs/pull/2771/merge:refs/remotes/origin/pr/2771/merge
fatal: Couldn't find remote ref refs/pull/2771/merge
[omr] Running shell script
+ curl -u **** --pass **** https://api.github.com/repos/eclipse/omr/pulls/2771
[omr] Running shell script
+ git fetch --progress origin +refs/heads/master:refs/remotes/origin/master
[omr] Running shell script
+ git checkout refs/remotes/origin/master
You are in 'detached HEAD' state.
HEAD is now at 31d44e1... Merge pull request #2874 from jdmpapin/icce-aliasing
```

**Comment by AdamBrousseau** [comment]
jenkins compile plinux jdk10 depends eclipse/omr#2881

**Comment by AdamBrousseau** [comment]
Test case: Depends on an open PR
```
Look for Dependent Changes
PullRequest source repo appears to be OpenJ9
Dependent OMR change found: #2881
+ bash ./get_source.sh
[2018-08-22 14:33:25] Get OpenJ9 sources
Clone repository: omr
Clone repository: openj9
[2018-08-22 14:33:38] OpenJ9 clone repositories finished in 13 seconds
[openj9] Running shell script
+ git fetch --progress origin +refs/pull/2415/merge:refs/remotes/origin/pr/2415/merge
 * [new ref]         refs/pull/2415/merge -> origin/pr/2415/merge
[openj9] Running shell script
+ git checkout refs/remotes/origin/pr/2415/merge
You are in 'detached HEAD' state.
HEAD is now at f64bea0... Merge 9d994e6e31f92170c3fcc7ff37fb243d60df880e into 767109edbe9fd2003476d65cab3a7d30a1ec943d
[omr] Running shell script
+ git config remote.origin.url https://github.com/eclipse/omr.git
[omr] Running shell script
+ git fetch --progress origin +refs/pull/2881/merge:refs/remotes/origin/pr/2881/merge
 * [new ref]         refs/pull/2881/merge -> origin/pr/2881/merge
[omr] Running shell script
+ git checkout refs/remotes/origin/pr/2881/merge
You are in 'detached HEAD' state.
HEAD is now at 2c06c8f... Merge a8e3c8e79393ece4e75ebb955be70348ef465ffa into 31d44e18ee711b11a750755e81d6f9bc7f2a9806
```

File: buildenv/jenkins/common/build [hunk 12]
```diff
@@ -195,26 +200,50 @@ def checkout_pullrequest() {
...
 
-def checkout_pullrequest(ID) {
+def checkout_pullrequest(ID, REPO) {
     // Determine if we are checking out a PR or a branch
     try {
         if (ID.isNumber()) {
-            sh "git fetch --progress origin +refs/pull/${ID}/merge:refs/remotes/origin/pr/${ID}/merge"
-            sh "git checkout refs/remotes/origin/pr/${ID}/merge"
+            try {
+                //sh "git fetch --progress origin +refs/pull/${ID}/merge:refs/remotes/origin/pr/${ID}/merge"
```
Comment (llxia) [inline buildenv/jenkins/common/build]: If this is no longer needed, can we remove it?

Comment (MichelleXiaoxiaoWang) [reply buildenv/jenkins/common/build]: fixed
